### ISSUE TITLE
`space-before-function-paren`: unable to resolve function members with generic parameters in the class

### ISSUE THREAD

**Issue by pawover** [eslint-stylistic/eslint-stylistic#1202]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

<img width="665" height="227" alt="Image" src="https://github.com/user-attachments/assets/7f74b9ee-a1b6-48c0-9d31-d43a1822ee5b" />

### Reproduction

screenshots are provided in the description

### System Info

```shell
"stylistic/space-before-function-paren": [2, { anonymous: "always", named: "always", asyncArrow: "always", catch: "always" }],
"eslint": "^10.2.1",
"@stylistic/eslint-plugin": "^5.10.0",
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
